src/main/java/com/squareup/kotlinpoet/FileSpec.kt

```diff
@@ -271,7 +272,15 @@ class FileSpec private constructor(builder: FileSpec.Builder) {
       this.indent = indent
     }
 
-    fun build() = FileSpec(this)
+    fun build(): FileSpec {
+      annotations.forEach { annotationSpec ->
+        if (annotationSpec.useSiteTarget != FILE) {
+          error(
+              "Use-site target ${annotationSpec.useSiteTarget} not supported for file annotations.")
```

Comment:
I don't follow this. Is this a new constraint?

Reply:
Technically it's not. In `addAnnotation`, if there is no site target, it synthesizes this file site target onto it. If there is one present and it's not file, it calls `error`. The final result is the same though, and the idea if that this check enforces that same requirement